src/Sylius/Bundle/ApiBundle/StateProcessor/Admin/Product/RemoveProcessor.php

```diff
@@ -31,13 +32,13 @@ public function __construct(
     /**
      * @param ProductInterface $data
      */
-    public function process(mixed $data, Operation $operation, array $uriVariables = [], array $context = [])
+    public function process(mixed $data, Operation $operation, array $uriVariables = [], array $context = []): void
     {
         Assert::isInstanceOf($data, ProductInterface::class);
         Assert::isInstanceOf($operation, DeleteOperationInterface::class);
 
         try {
-            return $this->removeProcessor->process($data, $operation, $uriVariables, $context);
+            $this->removeProcessor->process($data, $operation, $uriVariables, $context);
         } catch (ForeignKeyConstraintViolationException) {
             throw new ProductCannotBeRemoved();
         }
```

Comment:
We really should take a look if this can be abstracted out and somehow caught globally, so we won't need to create a try-catch for every resource for which this might occur. Could also ease end-app development.